add-on/chrome/test/mochitest/browser_throttler.js

```diff
@@ -0,0 +1,167 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
```

Comment:
For bonus points, port this to xpcshell so that it runs faster.  I don't think we want to block on that if it's non-trivial, though, so at least put an XXX comment suggesting that in here.